users/tests/test_logout_endpoint.py~

```diff
@@ -0,0 +1,56 @@
+from rest_framework.test import APITestCase
+from django.contrib.auth import get_user_model
+from rest_framework_simplejwt.token_blacklist.models import BlacklistedToken, OutstandingToken
+from rest_framework import status
+from users.models import UserRole
+
+User = get_user_model()
+
+class JWTLogoutTest(APITestCase):
+    def setUp(self):
+        # Create the role if it doesn't exist
+        self.role, _ = UserRole.objects.get_or_create(role="user")
+
+        # Create a user without a username field
+        self.user = User.objects.create_user(
+            email='test_user@example.com',
+            password='test_password123',
+            first_name='Test',
+            last_name='User',
+            role=self.role
+        )
+
+        self.login_url = '/api/users/auth/jwt/create/'
+        self.logout_url = '/api/users/auth/jwt/logout/'
+        self.refresh_url = '/api/users/auth/jwt/refresh/'
+
+        # Получаем токены
+        response = self.client.post(self.login_url, {
+            'email': 'test_user@example.com',
+            'password': 'test_password123'
+        })
+        self.assertEqual(response.status_code, 200, f"Login failed: {response.data}")
+
+        self.refresh_token = response.data['refresh']
+        self.access_token = response.data['access']
+
+    def test_logout_blacklists_refresh_token(self):
+        response = self.client.post(self.logout_url, {
+            'refresh': self.refresh_token
+        }, format='json')
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+        token = OutstandingToken.objects.get(token=self.refresh_token)
+        self.assertTrue(BlacklistedToken.objects.filter(token=token).exists())
+
+    def test_blacklisted_token_cannot_be_used(self):
+        self.client.post(self.logout_url, {'refresh': self.refresh_token}, format='json')
+
+        response = self.client.post(self.refresh_url, {
+            'refresh': self.refresh_token
+        }, format='json')
+
+        self.assertEqual(response.status_code, 401)
+        self.assertIn('token is blacklisted', str(response.data).lower())
```

Comment:
In line 55, instead of str(response.data).lower(), prefer response.data.get('detail') or similar for more robust error checking.
Add an assertion for response content-type (should be application/json).
Consider testing unblacklisting (if supported) or multiple blacklist attempts for completeness.